libraries/rush-lib/src/schemas/subspaces.schema.json

```diff
@@ -0,0 +1,34 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "Rush subspace config file.",
+  "description": "The configuration file for enabling the subspaces feature in rush. This is an EXPERIMENTAL feature which is not yet fully implemented. To opt into the experiemnt, simply toggle the 'enabled' property in this file.",
+  "type": "object",
+
+  "properties": {
+    "$schema": {
+      "description": "Part of the JSON Schema standard, this optional keyword declares the URL of the schema that the file conforms to. Editors may download the schema and use it to perform syntax highlighting.",
+      "type": "string"
+    },
+    "enabled": {
+      "description": "If true, rush will use the subspaces configuration",
+      "type": "boolean"
+    },
+    "splitWorkspaceCompatibility": {
+      "description": "(DEPRECATED) A depreciated option that allows individual subspaces to be configured at the package level if that package is the only project in the subspace. Used to help migrate from a split workspace state.",
+      "type": "boolean"
+    },
+    "availableSubspaces": {
```

Comment:
I could imagine that we might want to configure additional properties for a subspace in the future, although each subspace already gets its own folder under `common/config/subspaces/<subspace-name>/` so we could easily add config files there.

The main purpose for listing them in **subspaces.json** was to avoid having to enumerate `common/config/subspaces/*`, which has a few benefits:

- The `CODEOWNERS` for **subspaces.json** can be used to intercept people trying to add new subspaces
- A subspace can be temporarily removed without having to delete all its files
- If a previous Git checkout leaves behind some old files in **common/config/subspaces**, Rush won't try to process them

Reply:
Changed to just use string array for the subspace names, I think it does make sense to keep any future subspace configuration files in the individual config folder